### ISSUE TITLE
Proposal: simplified AssumeRolePolicyDocument

### ISSUE THREAD

**Issue by timoschilling** [aws/serverless-application-model#1573]
**Description:**

As far as I know `AWS::Serverless::Function.AssumeRolePolicyDocument` has only one use pattern

```yaml
AssumeRolePolicyDocument:
  Version: 2012-10-17
  Statement:
    - Effect: Allow
      Principal:
        Service:
          - <service>.amazonaws.com
          - <service>.amazonaws.com
      Action:
        - sts:AssumeRole
```

It would be very nice to shorten this to
```yaml
AssumeRolePolicyDocument:
  Service:
    - <service>.amazonaws.com
    - <service>.amazonaws.com
```

**Comment by aahung** [comment]
Thanks for the proposal.

> As far as I know AWS::Serverless::Function.AssumeRolePolicyDocument has only one use pattern

I don't think this is true, this assume role policy document will be attached to the role created for the function, and it be complicated.

For example:
- One might also want to let other AWS account to assume this role, then they can use `AWS: xxxxx` as the principal.
- On top of the previous use case, one might want to limit the IP address of the assume request, like [this](https://aws.amazon.com/premiumsupport/knowledge-center/iam-restrict-calls-ip-addresses/)

By preserving the existing format (which is provided by AWS IAM) allows such use cases, closing before we have an idea that can simplify without taking away functionalities.
